core/store/src/trie/resharding.rs

```diff
@@ -279,6 +344,85 @@ fn apply_delayed_receipts_to_children_states_impl(
     Ok(())
 }
 
+#[cfg(feature = "yield_resume")]
+fn apply_promise_yield_timeouts_to_children_states_impl(
```

Comment:
Not a big fan that we are essentially duplicating the `apply_delayed_receipts_to_children_states_impl` function here but I guess it's fine.

In the future, we would anyway most probably have to change the resharding implementation to take into account stateless validation and we can think about cleaning up then.

Reply:
I agree, the logic is perfectly equivalent between delayed receipts and promise yields and I thought a bit about writing something generic to handle both. I think we should consider it if we end up adding a third kind of queue which needs the same handling.

Reply:
Yeah, I don't think we should worry about it too much right now